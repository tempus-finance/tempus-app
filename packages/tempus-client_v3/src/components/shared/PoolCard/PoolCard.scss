.tc__poolCard {
  z-index: 0;
  position: relative;
  display: flex;
  flex-direction: column;
  width: 352px;
  padding: 32px;
  box-sizing: border-box;
  border: 1px solid var(--poolCardBorder);
  border-radius: 0px 40px;
  overflow: hidden;

  &[data-cardVariant='markets'] {
    height: 360px;
  }

  &[data-cardVariant='portfolio'] {
    height: 480px;
  }
}

.tc__poolCard-protocol {
  display: flex;
  align-items: center;
  margin-top: 2px;

  div {
    margin-left: 8px;
  }
}

.tc__poolCard-info {
  display: flex;
  flex-direction: column;
  justify-content: center;
  flex-grow: 1;

  > div:not(:last-child) {
    margin-bottom: 20px;
  }

  .tc__poolCard-info-row {
    > div:first-child {
      margin-bottom: 4px;
    }
  }
}

.tc__poolCard-user-data {
  display: flex;
  align-items: center;
}
