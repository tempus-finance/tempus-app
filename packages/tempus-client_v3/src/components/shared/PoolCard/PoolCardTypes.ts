export type PoolCardVariant = 'markets' | 'portfolio';
export type PoolCardStatus = 'Fixed' | 'Boosted' | 'Inactive' | 'Disabled' | 'Matured';
