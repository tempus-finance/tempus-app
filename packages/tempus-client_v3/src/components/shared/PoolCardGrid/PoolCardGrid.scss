.tc__pool-card-grid__content {
  margin-top: 24px;
  margin-bottom: 40px;

  display: grid;
  grid-template-columns: 1fr 1fr 1fr;
  grid-template-rows: auto;
  column-gap: 72px;
  row-gap: 24px;
}

.tc__pool-card-grid__show-more {
  display: flex;
  justify-content: center;
  margin-top: -16px;
  margin-bottom: 40px;
}
