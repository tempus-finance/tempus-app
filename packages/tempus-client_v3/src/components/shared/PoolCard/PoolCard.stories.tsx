import { ComponentStory, ComponentMeta } from '@storybook/react';
import { Decimal } from 'tempus-core-services';

import PoolCard from './PoolCard';

export default {
  title: 'PoolCard',
  component: PoolCard,
  argTypes: {},
} as ComponentMeta<typeof PoolCard>;

const style = {
  display: 'flex',
  alignItems: 'center',
  justifyContent: 'center',
};

const Template: ComponentStory<typeof PoolCard> = args => (
  <div style={style}>
    <PoolCard {...args} />
  </div>
);

export const Primary = Template.bind({});
Primary.args = {
  ticker: 'ETH',
  protocol: 'lido',
  poolCardType: 'portfolio',
  poolCartStatus: 'Matured',
  apr: new Decimal(0.123),
  terms: [new Date(2022, 7, 2), new Date(2023, 3, 4)],
  aggregatedAPR: new Decimal(0.042),
  totalBalance: new Decimal(10186124),
  color: '#627EEA',
  multiplier: 2,
};
