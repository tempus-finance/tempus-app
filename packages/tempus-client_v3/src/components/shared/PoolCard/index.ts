import PoolCard from './PoolCard';

export default PoolCard;

export * from './PoolCardTypes';
