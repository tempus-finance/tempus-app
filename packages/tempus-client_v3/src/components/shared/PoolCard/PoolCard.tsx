import { FC, useCallback, useMemo } from 'react';
import { useTranslation } from 'react-i18next';
import { Chain, Decimal, DecimalUtils, prettifyProtocolName, ProtocolName, Ticker, ZERO } from 'tempus-core-services';
import { min } from 'date-fns';
import FormattedDate from '../FormattedDate';
import LoadingPlaceholder from '../LoadingPlaceholder';
import Logo from '../Logo';
import Typography from '../Typography';
import { PoolCardStatus, PoolCardVariant } from './PoolCardTypes';
import PoolCardRipples from './PoolCardRipples/PoolCardRipples';
import PoolCardFlag from './PoolCardFlag/PoolCardFlag';

import './PoolCard.scss';
import { useLocale } from '../../../hooks';

interface PoolCardProps {
  chain: Chain;
  ticker: Ticker;
  protocol: ProtocolName;
  poolCardVariant: PoolCardVariant;
  poolCardStatus: PoolCardStatus;
  terms: Date[];
  aprValues: (Decimal | undefined)[];
  color: string;
  multiplier?: number;
  totalBalance?: Decimal;
  poolAddresses: string[];
  cardsInGroup?: number;
  onClick: (
    chain: Chain,
    ticker: Ticker,
    protocol: ProtocolName,
    status: PoolCardStatus,
    poolAddresses: string[],
  ) => void;
}

const PoolCard: FC<PoolCardProps> = props => {
  const {
    chain,
    ticker,
    protocol,
    poolCardVariant,
    poolCardStatus,
    terms,
    aprValues,
    color,
    totalBalance,
    multiplier = 1,
    poolAddresses,
    cardsInGroup = 1,
    onClick,
  } = props;
  const { t } = useTranslation();
  const [locale] = useLocale();

  const earliestTerm = useMemo(() => min(terms), [terms]);

  const maxApr = useMemo(() => {
    let max: Decimal | undefined;
    aprValues.forEach(aprValue => {
      if (aprValue && aprValue.gte(max ?? ZERO)) {
        max = aprValue;
      }
    });
    return max;
  }, [aprValues]);

  const maxAprFormatted = useMemo(() => (maxApr ? DecimalUtils.formatPercentage(maxApr) : undefined), [maxApr]);

  const totalBalanceFormatted = useMemo(() => {
    if (!totalBalance) {
      return null;
    }
    return DecimalUtils.formatWithMultiplier(totalBalance, 2);
  }, [totalBalance]);

  const aprLabel = useMemo(() => {
    if (poolCardVariant === 'markets') {
      return aprValues.length === 1 ? t('PoolCard.fixedApr') : t('PoolCard.fixedAprUpTo');
    }

    return aprValues.length === 1 ? t('PoolCard.apr') : t('PoolCard.aprUpTo');
  }, [aprValues.length, poolCardVariant, t]);

  const handleClick = useCallback(() => {
    onClick(chain, ticker, protocol, poolCardStatus, poolAddresses);
  }, [chain, ticker, protocol, poolCardStatus, poolAddresses, onClick]);

  const loading = !maxApr;

  return (
    <div className="tc__poolCard-container" data-card-stack={cardsInGroup > 1 ? Math.min(cardsInGroup, 3) : undefined}>
      <div className="tc__poolCard" data-card-variant={poolCardVariant} onClick={handleClick}>
        {/* Pool backing token ticker */}
        {loading && (
          <div className="tc__poolCard-value-placeholder-container">
            <LoadingPlaceholder width="small" height="medium" />
          </div>
        )}
        {!loading && (
          <Typography variant="subheader" weight="bold">
            {multiplier > 1 && `x${multiplier}`} {ticker}
          </Typography>
        )}

        {/* Pool underlying protocol name */}
        {loading && (
          <div className="tc__poolCard-protocol-placeholder-container">
            <LoadingPlaceholder width="tiny" height="small" />
          </div>
        )}
        {!loading && (
          <div className="tc__poolCard-protocol">
            <Logo type={`protocol-${protocol}`} size={20} />
            <Typography variant="body-secondary" weight="medium">
              {prettifyProtocolName(protocol)}
            </Typography>
          </div>
        )}

        <div className="tc__poolCard-info">
          {/* APR - Shown only for non mature pools */}
          <div className="tc__poolCard-info-row">
            <Typography variant="body-secondary" weight="medium" color="text-secondary">
              {aprLabel}
            </Typography>
            {loading && (
              <div className="tc__poolCard-value-placeholder-container">
                <LoadingPlaceholder width="medium" height="medium" />
              </div>
            )}
            {!loading && (
              <Typography variant="subheader" weight="medium" type="mono">
                {poolCardStatus === 'Matured' ? t('PoolCard.notApplicable') : maxAprFormatted}
              </Typography>
            )}
          </div>

          {/* Term */}
          <div className="tc__poolCard-info-row">
            <Typography variant="body-secondary" weight="medium" color="text-secondary">
              {terms.length > 1 ? t('PoolCard.earliestTerm') : t('PoolCard.term')}
            </Typography>
            {loading && (
              <div className="tc__poolCard-value-placeholder-container">
                <LoadingPlaceholder width="large" height="medium" />
              </div>
            )}
            {!loading && <FormattedDate date={earliestTerm} locale={locale} size="large" />}
          </div>

          {/* Total Balance */}
          {poolCardVariant === 'portfolio' && totalBalanceFormatted && (
            <div className="tc__poolCard-info-row">
              <Typography variant="body-secondary" weight="medium" color="text-secondary">
                {cardsInGroup && cardsInGroup > 1 ? t('PoolCard.totalBalance') : t('PoolCard.balance')}
              </Typography>
              {loading && (
                <div className="tc__poolCard-value-placeholder-container">
                  <LoadingPlaceholder width="medium" height="medium" />
                </div>
              )}
              {!loading && (
                <Typography variant="subheader" weight="medium" type="mono">
                  {totalBalanceFormatted}
                </Typography>
              )}
            </div>
          )}
        </div>

        {!loading && (
          <>
            <PoolCardRipples color={color} />
            <PoolCardFlag ticker={ticker} status={poolCardStatus} />
          </>
        )}
      </div>

      {cardsInGroup >= 2 && <div className="tc__poolCard-stack-card tc__poolCard-stack-card-2" />}
      {cardsInGroup >= 3 && <div className="tc__poolCard-stack-card tc__poolCard-stack-card-3" />}
    </div>
  );
};
export default PoolCard;
