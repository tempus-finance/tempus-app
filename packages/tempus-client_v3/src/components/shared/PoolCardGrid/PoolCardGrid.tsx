import { FC, useCallback, useEffect, useState } from 'react';
import { useTranslation } from 'react-i18next';
import { Chain, Decimal, ProtocolName, TempusPool, Ticker, tokenColorMap } from 'tempus-core-services';
import { PoolCardVariant } from '../PoolCard/PoolCardTypes';
import { useFixedAprs } from '../../../hooks';
import ActionButton from '../ActionButton';
import PoolCard, { PoolCardStatus } from '../PoolCard';

import './PoolCardGrid.scss';

export interface PoolCardData {
  chain: Chain;
  token: Ticker;
  tokenAddress: string;
  protocol: ProtocolName;
  matured: boolean;
  totalBalance?: Decimal;
  pools: TempusPool[];
}

export interface PoolCardGridProps {
  chain: Chain;
  cards: PoolCardData[];
  cardVariant: PoolCardVariant;
  onCardClick: (
    chain: Chain,
    ticker: Ticker,
    protocol: ProtocolName,
    status: PoolCardStatus,
    poolAddresses: string[],
  ) => void;
}

export const NUMBER_OF_CARDS_PER_PAGE = 6;
export const NUMBER_OF_CARDS_SHOW_AT_START = 3;

const PoolCardGrid: FC<PoolCardGridProps> = props => {
  const { chain, cards, cardVariant, onCardClick } = props;
  const fixedAprs = useFixedAprs();
  const { t } = useTranslation();

  // When users clicks show more button, number of visible pools is increased
  // When user sorts/filters pool list - number of visible pools for each chain is reset
  const [visiblePools, setVisiblePools] = useState(NUMBER_OF_CARDS_SHOW_AT_START);

  // Reset number of visible pools when tempus pools list changes (ie. user filtered/sorted pools)
  useEffect(() => {
    setVisiblePools(NUMBER_OF_CARDS_SHOW_AT_START);
  }, [cards]);

  const onShowMoreClick = useCallback(() => {
    setVisiblePools(prevState => (prevState || NUMBER_OF_CARDS_SHOW_AT_START) + NUMBER_OF_CARDS_PER_PAGE);
  }, []);

  const cardsToShow = cards.slice(0, visiblePools || NUMBER_OF_CARDS_SHOW_AT_START);

  return (
    <div className="tc__pool-card-grid">
      <div className="tc__pool-card-grid__content">
        {cardsToShow.map(card => {
          const cardColor = tokenColorMap.get(card.token);
          if (!cardColor) {
            console.warn(`Missing ${card.token} token color in tokenColorMap!`);
          }

          const terms = card.pools.map(pool => new Date(pool.maturityDate));
          const poolAddresses = card.pools.map(pool => pool.address);
          const aprs = card.pools.map(pool => fixedAprs[`${pool.chain}-${pool.address}`]);

          let cardStatus: PoolCardStatus = 'Fixed';
          if (card.matured) {
            cardStatus = 'Matured';
          }

          return (
            <PoolCard
              key={`${chain}-${card.protocol}-${card.tokenAddress}-${card.matured}`}
              aprValues={aprs}
              color={cardColor || '#ffffff'}
              poolCardStatus={cardStatus}
              poolCardVariant={cardVariant}
              ticker={card.token}
              protocol={card.protocol}
              terms={terms}
              chain={card.chain}
              onClick={onCardClick}
              poolAddresses={poolAddresses}
              totalBalance={card.totalBalance}
            />
          );
        })}
      </div>
      {cardsToShow.length < cards.length && (
        <div className="tc__pool-card-grid__show-more">
          {/* show more button */}
          <ActionButton
            labels={{
              default:
                cards.length - cardsToShow.length > NUMBER_OF_CARDS_PER_PAGE
                  ? t('MarketsPools.showMoreXOfYButtonLabel', {
                      numOfCardsToShow: NUMBER_OF_CARDS_PER_PAGE,
                      numOfRemainingCards: cards.length - cardsToShow.length,
                    })
                  : t('MarketsPools.showMoreXButtonLabel', {
                      numOfCardsToShow: cards.length - cardsToShow.length,
                    }),
              loading: '',
              success: '',
            }}
            onClick={onShowMoreClick}
            variant="secondary"
            size="large"
          />
        </div>
      )}
    </div>
  );
};

export default PoolCardGrid;
