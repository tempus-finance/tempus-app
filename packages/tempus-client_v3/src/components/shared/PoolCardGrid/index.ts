import PoolCardGrid from './PoolCardGrid';

export default PoolCardGrid;
export * from './PoolCardGrid';
