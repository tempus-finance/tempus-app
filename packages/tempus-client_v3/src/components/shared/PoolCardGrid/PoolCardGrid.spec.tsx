import { fireEvent, render } from '@testing-library/react';
import { Chain, ProtocolName, Ticker } from 'tempus-core-services';
import { pool3, pool4, pool5 } from '../../../setupTests';
import { PoolCardStatus } from '../PoolCard';
import PoolCardGrid, {
  NUMBER_OF_CARDS_PER_PAGE,
  NUMBER_OF_CARDS_SHOW_AT_START,
  PoolCardData,
  PoolCardGridProps,
} from './PoolCardGrid';

const mockOnCardClick = jest.fn<void, [Chain, Ticker, ProtocolName, PoolCardStatus, string[]]>();

const dateNow = new Date(2023, 0, 1);

const defaultProps: PoolCardGridProps = {
  chain: 'fantom',
  // Pools are duplicated for testing "show more" button
  cards: [pool3, pool4, pool5, pool3, pool4, pool5, pool3, pool4, pool5, pool3].map(
    pool =>
      ({
        chain: pool.chain,
        token: pool.backingToken,
        tokenAddress: pool.backingTokenAddress,
        protocol: pool.protocol,
        matured: pool.maturityDate <= dateNow.getTime(),
        pools: [pool],
      } as PoolCardData),
  ),
  cardVariant: 'markets',
  onCardClick: mockOnCardClick,
};

const subject = (props: PoolCardGridProps) => render(<PoolCardGrid {...props} />);

describe('PoolCardGrid', () => {
  beforeEach(() => {
    // React complains because some cards have same `key` props. That happens as
    // there are duplicated pools in this test. In real world scenario, that
    // won't happen.
    jest.spyOn(console, 'error').mockImplementation();
  });

  it('renders a pool card grid', () => {
    const { container } = subject(defaultProps);

    expect(container).not.toBeNull();
    expect(container).toMatchSnapshot();
  });

  it('calls `onClick` when a pool card is clicked', () => {
    const { container } = subject(defaultProps);

    const poolCards = container.querySelectorAll('.tc__poolCard');

    expect(poolCards).toHaveLength(NUMBER_OF_CARDS_SHOW_AT_START);

    poolCards.forEach((poolCard, index) => {
      fireEvent.click(poolCard);

      const cardData = defaultProps.cards[index];

      expect(mockOnCardClick).toHaveBeenCalledTimes(index + 1);
      expect(mockOnCardClick).toHaveBeenCalledWith(
        cardData.chain,
        cardData.token,
        cardData.protocol,
        cardData.matured ? 'Matured' : 'Fixed',
        cardData.pools.map(pool => pool.address),
      );
    });
  });

  it('shows more cards when "Show more" button is clicked', () => {
    const { container } = subject(defaultProps);

    let poolCards = container.querySelectorAll('.tc__poolCard');

    expect(poolCards).toHaveLength(NUMBER_OF_CARDS_SHOW_AT_START);

    let showMoreButton = container.querySelector('.tc__actionButton');

    expect(showMoreButton).not.toBeNull();
    expect(showMoreButton).toHaveTextContent(
      `Show ${NUMBER_OF_CARDS_PER_PAGE} of ${defaultProps.cards.length - NUMBER_OF_CARDS_SHOW_AT_START}`,
    );
    expect(showMoreButton).toMatchSnapshot();

    fireEvent.click(showMoreButton as Element);

    poolCards = container.querySelectorAll('.tc__poolCard');
    const numberOfShownCards = NUMBER_OF_CARDS_SHOW_AT_START + NUMBER_OF_CARDS_PER_PAGE;

    expect(poolCards).toHaveLength(numberOfShownCards);

    expect(showMoreButton).toHaveTextContent(`Show ${defaultProps.cards.length - numberOfShownCards} more`);
    expect(showMoreButton).toMatchSnapshot();

    fireEvent.click(showMoreButton as Element);

    poolCards = container.querySelectorAll('.tc__poolCard');
    showMoreButton = container.querySelector('.tc__actionButton');

    expect(poolCards).toHaveLength(defaultProps.cards.length);
    expect(showMoreButton).toBeNull();
  });
});
