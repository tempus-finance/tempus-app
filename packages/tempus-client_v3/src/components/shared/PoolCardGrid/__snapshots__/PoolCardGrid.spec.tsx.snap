// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PoolCardGrid renders a pool card grid 1`] = `
<div>
  <div
    class="tc__pool-card-grid"
  >
    <div
      class="tc__pool-card-grid__content"
    >
      <div
        class="tc__poolCard"
        data-card-variant="markets"
      >
        <div
          class="tc__poolCard-value-placeholder-container"
        >
          <div
            class="tc__loading-placeholder tc__loading-placeholder__width-small tc__loading-placeholder__height-medium"
          >
            <div
              class="tc__loading-placeholder__circle"
            />
            <div
              class="tc__loading-placeholder__bar"
            />
          </div>
        </div>
        <div
          class="tc__poolCard-protocol-placeholder-container"
        >
          <div
            class="tc__loading-placeholder tc__loading-placeholder__width-tiny tc__loading-placeholder__height-small"
          >
            <div
              class="tc__loading-placeholder__circle"
            />
            <div
              class="tc__loading-placeholder__bar"
            />
          </div>
        </div>
        <div
          class="tc__poolCard-info"
        >
          <div
            class="tc__poolCard-info-row"
          >
            <div
              style="font-style: normal; font-size: 12px; line-height: 16px; color: rgb(71, 67, 80); font-weight: 500; font-family: 'DM Sans', sans-serif; opacity: 1;"
            >
              Fixed APR
            </div>
            <div
              class="tc__poolCard-value-placeholder-container"
            >
              <div
                class="tc__loading-placeholder tc__loading-placeholder__width-medium tc__loading-placeholder__height-medium"
              >
                <div
                  class="tc__loading-placeholder__circle"
                />
                <div
                  class="tc__loading-placeholder__bar"
                />
              </div>
            </div>
          </div>
          <div
            class="tc__poolCard-info-row"
          >
            <div
              style="font-style: normal; font-size: 12px; line-height: 16px; color: rgb(71, 67, 80); font-weight: 500; font-family: 'DM Sans', sans-serif; opacity: 1;"
            >
              Term
            </div>
            <div
              class="tc__poolCard-value-placeholder-container"
            >
              <div
                class="tc__loading-placeholder tc__loading-placeholder__width-large tc__loading-placeholder__height-medium"
              >
                <div
                  class="tc__loading-placeholder__circle"
                />
                <div
                  class="tc__loading-placeholder__bar"
                />
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="tc__poolCard"
        data-card-variant="markets"
      >
        <div
          class="tc__poolCard-value-placeholder-container"
        >
          <div
            class="tc__loading-placeholder tc__loading-placeholder__width-small tc__loading-placeholder__height-medium"
          >
            <div
              class="tc__loading-placeholder__circle"
            />
            <div
              class="tc__loading-placeholder__bar"
            />
          </div>
        </div>
        <div
          class="tc__poolCard-protocol-placeholder-container"
        >
          <div
            class="tc__loading-placeholder tc__loading-placeholder__width-tiny tc__loading-placeholder__height-small"
          >
            <div
              class="tc__loading-placeholder__circle"
            />
            <div
              class="tc__loading-placeholder__bar"
            />
          </div>
        </div>
        <div
          class="tc__poolCard-info"
        >
          <div
            class="tc__poolCard-info-row"
          >
            <div
              style="font-style: normal; font-size: 12px; line-height: 16px; color: rgb(71, 67, 80); font-weight: 500; font-family: 'DM Sans', sans-serif; opacity: 1;"
            >
              Term
            </div>
            <div
              class="tc__poolCard-value-placeholder-container"
            >
              <div
                class="tc__loading-placeholder tc__loading-placeholder__width-large tc__loading-placeholder__height-medium"
              >
                <div
                  class="tc__loading-placeholder__circle"
                />
                <div
                  class="tc__loading-placeholder__bar"
                />
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="tc__poolCard"
        data-card-variant="markets"
      >
        <div
          class="tc__poolCard-value-placeholder-container"
        >
          <div
            class="tc__loading-placeholder tc__loading-placeholder__width-small tc__loading-placeholder__height-medium"
          >
            <div
              class="tc__loading-placeholder__circle"
            />
            <div
              class="tc__loading-placeholder__bar"
            />
          </div>
        </div>
        <div
          class="tc__poolCard-protocol-placeholder-container"
        >
          <div
            class="tc__loading-placeholder tc__loading-placeholder__width-tiny tc__loading-placeholder__height-small"
          >
            <div
              class="tc__loading-placeholder__circle"
            />
            <div
              class="tc__loading-placeholder__bar"
            />
          </div>
        </div>
        <div
          class="tc__poolCard-info"
        >
          <div
            class="tc__poolCard-info-row"
          >
            <div
              style="font-style: normal; font-size: 12px; line-height: 16px; color: rgb(71, 67, 80); font-weight: 500; font-family: 'DM Sans', sans-serif; opacity: 1;"
            >
              Fixed APR
            </div>
            <div
              class="tc__poolCard-value-placeholder-container"
            >
              <div
                class="tc__loading-placeholder tc__loading-placeholder__width-medium tc__loading-placeholder__height-medium"
              >
                <div
                  class="tc__loading-placeholder__circle"
                />
                <div
                  class="tc__loading-placeholder__bar"
                />
              </div>
            </div>
          </div>
          <div
            class="tc__poolCard-info-row"
          >
            <div
              style="font-style: normal; font-size: 12px; line-height: 16px; color: rgb(71, 67, 80); font-weight: 500; font-family: 'DM Sans', sans-serif; opacity: 1;"
            >
              Term
            </div>
            <div
              class="tc__poolCard-value-placeholder-container"
            >
              <div
                class="tc__loading-placeholder tc__loading-placeholder__width-large tc__loading-placeholder__height-medium"
              >
                <div
                  class="tc__loading-placeholder__circle"
                />
                <div
                  class="tc__loading-placeholder__bar"
                />
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="tc__pool-card-grid__show-more"
    >
      <button
        class="tc__btn tc__actionButton tc__actionButton-border-secondary-large tc__actionButton-background-secondary-large tc__actionButton-hover-secondary-large tc__actionButton-shadow-default tc__actionButton-cursor-default tc__actionButton-size-large "
        data-selected="false"
        type="button"
      >
        <div
          class="tc__actionButton-status-label"
        >
          <div
            style="font-style: normal; font-size: 16px; line-height: 24px; color: rgb(34, 34, 34); font-weight: 700; font-family: 'DM Sans', sans-serif; opacity: 1;"
          >
            Show 6 of 7 more
          </div>
        </div>
      </button>
    </div>
  </div>
</div>
`;

exports[`PoolCardGrid shows more cards when "Show more" button is clicked 1`] = `
<button
  class="tc__btn tc__actionButton tc__actionButton-border-secondary-large tc__actionButton-background-secondary-large tc__actionButton-hover-secondary-large tc__actionButton-shadow-default tc__actionButton-cursor-default tc__actionButton-size-large "
  data-selected="false"
  type="button"
>
  <div
    class="tc__actionButton-status-label"
  >
    <div
      style="font-style: normal; font-size: 16px; line-height: 24px; color: rgb(34, 34, 34); font-weight: 700; font-family: 'DM Sans', sans-serif; opacity: 1;"
    >
      Show 6 of 7 more
    </div>
  </div>
</button>
`;

exports[`PoolCardGrid shows more cards when "Show more" button is clicked 2`] = `
<button
  class="tc__btn tc__actionButton tc__actionButton-border-secondary-large tc__actionButton-background-secondary-large tc__actionButton-hover-secondary-large tc__actionButton-shadow-default tc__actionButton-cursor-default tc__actionButton-size-large "
  data-selected="false"
  type="button"
>
  <div
    class="tc__actionButton-status-label"
  >
    <div
      style="font-style: normal; font-size: 16px; line-height: 24px; color: rgb(34, 34, 34); font-weight: 700; font-family: 'DM Sans', sans-serif; opacity: 1;"
    >
      Show 1 more
    </div>
  </div>
</button>
`;
